### PR TITLE
Use params everywhere

### PR DESCRIPTION
**Pull request by camillobruni**
- Add support for filtering multiple suites
- Support startAutomatically in the default runner
- Fix displaying detailed results with unit=ms
- More error messages on invalid url-params
- Add Suites.enable function instead of two separate implementations per runner

### CODE REVIEW

File: resources/params.mjs [hunk 1]
```diff
@@ -15,19 +18,42 @@ class Params {
...
+
+        if (searchParams.has("unit")) {
+            this.unit = searchParams.get("unit");
+            if (!UNITS.includes(this.unit))
+                throw new Error(`Invalid unit=${this.unit}. Valid values are ${UNITS}`);
+            searchParams.delete("unit");
+        }
+
+        if (searchParams.has("suite") || searchParams.has("suites")) {
+            const value = searchParams.get("suites") || searchParams.get("suites");
```
Comment (rniwa) [inline resources/params.mjs:47]: Isn't one of these supposed to be "suite", and not both "suites"?

File: resources/tests.mjs [hunk 2]
```diff
@@ -3,6 +3,21 @@ import { BenchmarkTestStep } from "./benchmark-runner.mjs";
 const numberOfItemsToAdd = 100;
 export const Suites = [];
 
+Suites.enable = function (names) {
+    this.forEach((suite) => {
+        suite.disabled = true;
```
Comment (rniwa) [inline resources/tests.mjs:8]: It doesn't seem like this forEach is needed. We can just add else clause in the forEach call below.

File: resources/tests.mjs [hunk 3]
```diff
@@ -443,3 +458,8 @@ Suites.push({
         }),
     ],
 });
+
+Object.freeze(Suites);
```
Comment (rniwa) [inline resources/tests.mjs:462]: Let's not do this in this PR?

File: resources/params.mjs (hunk 1 above)
Comment (camillobruni) [reply resources/params.mjs:47]: - I kept "suite" for backwards compatiblity (mayabe that's adding more confusion?)
- I'll add a warning if both "suite" and suites" are specified

Comment (rniwa) [reply resources/params.mjs:47]: What I mean is that you have "suites" or "suites". Both of them are suites instead of one being "suite".

Comment (camillobruni) [reply resources/params.mjs:47]: Ah sorry, misread. DONE.

File: resources/tests.mjs (hunk 3 above)
Comment (camillobruni) [reply resources/tests.mjs:462]: removed.

File: resources/tests.mjs (hunk 2 above)
Comment (camillobruni) [reply resources/tests.mjs:8]: DONE

File: resources/main.mjs [hunk 4]
```diff
@@ -19,51 +19,20 @@ class MainBenchmarkClient {
...
-                        }
-                        break;
-                }
+        this.displayUnit = params.unit;
+        this.iterationCount = params.iterationCount;
+        if (params.suites.length > 0) {
+            if (!Suites.enable(params.suites)) {
+                const message = `Suite "${params.suites}" does not exist. No tests to run.`;
+                alert(message);
+                console.error(
```
Comment (rniwa) [inline resources/main.mjs:29]: It seems that this console.error could fit in a single line?

**Review by rniwa — APPROVED** [review]

File: resources/main.mjs (hunk 4 above)
Comment (camillobruni) [reply resources/main.mjs:29]: Double checked, `npm run format` somehow want's this on multiple lines :(.

Comment (rniwa) [reply resources/main.mjs:29]: Weird.

File: resources/params.mjs [hunk 5]
```diff
@@ -15,19 +18,43 @@ class Params {
...
+        searchParams.delete("startAutomatically");
+        if (searchParams.has("iterationCount")) {
+            this.iterationCount = parseInt(searchParams.get("iterationCount")) || this.iterationCount;
+            if (this.iterationCount <= 1)
+                throw new Error(`Invalid iterationCount param: ${this.iterationCount}`);
+            searchParams.delete("iterationCount");
+        }
+
+        if (searchParams.has("unit")) {
+            this.unit = searchParams.get("unit");
```
[the remark this answers is not in the document]
Comment (camillobruni) [reply resources/params.mjs]: done.

**Pull request merged**